tests/integration/targets/dns_delegation/tasks/main.yml

```diff
@@ -0,0 +1,243 @@
+            fqdn: "ns1.com."
+        state: present
+      register: delegation
+    - name: Get Information about the DNS Delegation
+      infoblox.bloxone.dns_delegation_info:
+        filters:
+          fqdn: "{{ delegation_fqdn }}"
+      register: delegation_info
+    - assert:
+        that:
+          - delegation_info is not failed
+          - delegation_info.objects | length == 1
+          - delegation_info.objects[0].id == delegation.id
+          - delegation_info.objects[0].disabled == false
+          - delegation_info.objects[0].comment == "Delegation zone is created by Ansible"
+
+    - name: Create DNS Delegation with FQDN
+      infoblox.bloxone.dns_delegation:
+        fqdn: "{{ delegation_fqdn }}"
+        comment: "Delegation zone created with Ansible for FQDN test"
+        delegation_servers:
+          - address: "12.0.0.0"
+            fqdn: "ns1.com."
+        state: present
+      register: delegation
+    - name: Get Information about the DNS Delegation
+      infoblox.bloxone.dns_delegation_info:
+        filters:
+          fqdn: "{{ delegation_fqdn }}"
+      register: delegation_info
+    - assert:
+        that:
+          - delegation_info is not failed
+          - delegation_info.objects | length == 1
+          - delegation_info.objects[0].id == delegation.id
+          - delegation_info.objects[0].fqdn == delegation_fqdn
+          - delegation_info.objects[0].comment == "Delegation zone created with Ansible for FQDN test"
+
+    - name: Create DNS Delegation with Tags
+      infoblox.bloxone.dns_delegation:
+        fqdn: "{{ delegation_fqdn }}"
+        comment: "Delegation zone created with Ansible for tags test"
+        delegation_servers:
+          - address: "12.0.0.0"
+            fqdn: "ns1.com."
+        tags:
+          location: "site-1"
+        state: present
+      register: delegation
+    - name: Get Information about the DNS Delegation
+      infoblox.bloxone.dns_delegation_info:
+        filters:
+          fqdn: "{{ delegation_fqdn }}"
+      register: delegation_info
+    - assert:
+        that:
+          - delegation_info is not failed
+          - delegation_info.objects | length == 1
+          - delegation_info.objects[0].id == delegation.id
+          - delegation_info.objects[0].fqdn == delegation_fqdn
+          - delegation_info.objects[0].comment == "Delegation zone created with Ansible for tags test"
+          - delegation_info.objects[0].tags.location == "site-1"
+          - delegation_info.objects[0].view == _view.id
+
+  always:
+    - name: "Clean up DNS Delegation"
+      infoblox.bloxone.dns_delegation:
+        fqdn: "{{ delegation_fqdn }}"
+        state: absent
+      ignore_errors: true
+
+    - name: "Clean up Authorization Zone"
+      infoblox.bloxone.dns_auth_zone:
+        fqdn: "{{ _fqdn_auth_zone }}"
+        state: absent
+      ignore_errors: true
+
+    - name: "Clean up  DNS View"
+      infoblox.bloxone.dns_view:
+        name: "{{ _view.id }}"
```

Comment:
Why is view name being assigned to Id here ?

Reply:
corrected this and tested